@here/harp-examples/src/threejs_shadows.ts

```diff
@@ -0,0 +1,265 @@
+import { isJsonExpr } from "@here/harp-datasource-protocol/lib/Expr";
+import { GeoCoordinates } from "@here/harp-geoutils";
+import { MapControls, MapControlsUI } from "@here/harp-map-controls";
+import {
+    CopyrightElementHandler,
+    MapView,
+    MapViewEventNames,
+    ThemeLoader
+} from "@here/harp-mapview";
+import { APIFormat, AuthenticationMethod, OmvDataSource } from "@here/harp-omv-datasource";
+import { assert } from "@here/harp-utils";
+import { GUI } from "dat.gui";
+import * as THREE from "three";
+import "three/examples/js/controls/TrackballControls";
+import { ShadowMapViewer } from "three/examples/jsm/utils/ShadowMapViewer";
+import { apikey, copyrightInfo } from "../config";
+
+let map: MapView;
+
+const guiOptions = {
+    xpos: 0,
+    ypos: 0,
+    zpos: 0,
+    enabled: true
+};
+
+const swapCamera = (
+    mapControls: MapControls,
+    trackBall: any,
+    debugCamera: THREE.PerspectiveCamera
+) => {
+    mapControls.enabled = !mapControls.enabled;
+    trackBall.enabled = !trackBall.enabled;
+    map.pointOfView = mapControls.enabled ? undefined : debugCamera;
+};
+
+const setupDebugCamera = (mapControls: MapControls) => {
+    // tslint:disable-next-line: no-string-literal
+    const mapCameraHelper = new THREE.CameraHelper(map["m_rteCamera"]);
+    mapCameraHelper.renderOrder = Number.MAX_SAFE_INTEGER;
+    map.scene.add(mapCameraHelper);
+
+    const debugCamera = new THREE.PerspectiveCamera(
+        map.camera.fov,
+        map.canvas.width / map.canvas.height,
+        100,
+        100000
+    );
+    map.scene.add(debugCamera);
+    debugCamera.position.set(6000, 2000, 1000);
+
+    const m_trackball = new (THREE as any).TrackballControls(debugCamera, map.canvas);
+    m_trackball.enabled = false;
+    m_trackball.addEventListener("start", () => {
+        map.beginAnimation();
+    });
+    m_trackball.addEventListener("end", () => {
+        map.endAnimation();
+    });
+    m_trackball.addEventListener("change", () => {
+        map.update();
+    });
+
+    m_trackball.staticMoving = true;
+    m_trackball.rotateSpeed = 3.0;
+    m_trackball.zoomSpeed = 4.0;
+    m_trackball.panSpeed = 2.0;
+
+    let lastZoomLevel = map.zoomLevel;
+    // Update the debug controls before rendering.
+    map.addEventListener(MapViewEventNames.Render, () => {
+        if (m_trackball !== undefined) {
+            const trackballTarget = m_trackball.target as THREE.Vector3;
+            if (lastZoomLevel !== map.zoomLevel) {
+                trackballTarget.set(0, 0, -map.targetDistance);
+                lastZoomLevel = map.zoomLevel;
+            }
+            m_trackball.update();
+        }
+        const enableCameraHelpers = map.pointOfView !== undefined;
+        if (enableCameraHelpers) {
+            mapCameraHelper.update();
+        }
+        mapCameraHelper.visible = enableCameraHelpers;
+    });
+
+    window.addEventListener("keypress", event => {
+        if (event.key === "s") {
```

Comment:
can we have sth. that also works on the phone?

Reply:
Good point, I will add to the gui.